### ISSUE TITLE
Misleading Tuple type

### ISSUE THREAD

**Issue by python-app-generation** [maximkulkin/lollipop#52]
Is it on purpose that the Tuple type dumps into / loads from a list instead of a tuple? If yes, I think the naming is a bit misleading…

```
Tuple([String(), Integer(), Boolean()]).load(('foo', 123, False))
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "site-packages\lollipop\types.py", line 443, in load self._fail('invalid')
  File "site-packages\lollipop\errors.py", line 63, in _fail raise ValidationError(msg)
lollipop.errors.ValidationError: Invalid data: 'Value should be list'
```

**Comment by maximkulkin** [comment]
The idea of tuple is to allow tuple on "application" (the other) side. It dumps to/loads from a list:
```python
t = Tuple([String(), Integer(), Boolean()])
t.dump(('foo', 123, False))
# => ['foo', 123, False]

t.load(['bar', 456, True])
# => ('bar', 456, True)
```

**Comment by python-app-generation** [comment]
Sorry, I should have used dump instead of load in my example. It does not work for me:

```
t = Tuple([String(), Integer(), Boolean()])
t.dump(('foo', 123, False))
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File ".../site-packages/lollipop/types.py", line 464, in dump
    self._fail('invalid')
  File ".../site-packages/lollipop/errors.py", line 63, in _fail
    raise ValidationError(msg)
lollipop.errors.ValidationError: Invalid data: 'Value should be list'
```

I guess the problem is this check in Tuple.dump:

```
if not is_list(value):
            self._fail('invalid')

```

I'm using 1.1.3 with Python 3.6.1.

**Comment by maximkulkin** [comment]
Indeed, this is an overlook on my side. Fixing it

**Comment by maximkulkin** [comment]
Should be fixed now in v1.1.4
